client/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+#![no_std]
+
+use core::fmt::Debug;
+use core::marker::PhantomData;
+use gstd::{msg::MessageFuture, prelude::*, ActorId, MessageId};
+use parity_scale_codec::{Decode, Encode};
+
+#[non_exhaustive]
+#[derive(Default)]
+pub struct SendArgs {
+    value: u128,
+    _sender_id: [u8; 32],
+}
+
+#[non_exhaustive]
+pub struct SendTicket<R: Decode + Debug> {
+    // message we are waiting on
+    f: MessageFuture,
+    // silence the compiler
+    _marker: PhantomData<R>,
+}
+
+impl<R: Decode + Debug> SendTicket<R> {
+    pub async fn result(self) -> Result<R, gstd::errors::Error> {
+        let payload = self.f.await?;
+
+        R::decode(&mut payload.as_ref()).map_err(gstd::errors::Error::Decode)
+    }
+
+    pub fn message_id(&self) -> MessageId {
+        self.f.waiting_reply_to
+    }
+}
+
+/// Sender that runs message against gtest::Program
+#[derive(Default)]
+pub struct NativeSender;
+impl NativeSender {
+    pub fn new() -> Self {
+        Self
+    }
+}
+
+#[derive(Default)]
+pub struct Call<R: Decode + Debug> {
+    /// request payload
+    payload: Vec<u8>,
+    /// optional args
+    args: SendArgs,
+    /// silence the compiler
+    _marker: PhantomData<R>,
+}
+
+#[non_exhaustive]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum SendError {
+    Parse(parity_scale_codec::Error),
+    Sender(gstd::errors::Error),
+}
+
+impl From<parity_scale_codec::Error> for SendError {
+    fn from(e: parity_scale_codec::Error) -> Self {
+        Self::Parse(e)
+    }
+}
+
+impl From<gstd::errors::Error> for SendError {
+    fn from(e: gstd::errors::Error) -> Self {
+        Self::Sender(e)
+    }
+}
+
+impl<R: Decode + Debug> Call<R> {
+    pub fn new(payload: Vec<u8>) -> Self {
+        Self {
+            payload,
+            args: Default::default(),
+            _marker: PhantomData,
+        }
+    }
+
+    pub async fn send(self, address: impl Into<[u8; 32]>) -> Result<SendTicket<R>, SendError> {
```

Comment:
```suggestion
    pub async fn send(self, dest: ActorId) -> Result<SendTicket<R>, SendError> {
```

Reply:
Changed, but my vision here was to abstract all libraries so `Call` is universal across all environments. Using `gstd::ActorId` requires importing `gstd`, which as far as I understand isn't compatible with `std`